src/ray/rpc/client_call.h

```diff
@@ -195,6 +195,11 @@ class ClientCallManager {
     for (auto &polling_thread : polling_threads_) {
       polling_thread.join();
     }
+    // gRPC's completion queue won't clear our call tags, we need to clear call tags that
```

Comment:
IIUC, this leak only affects shutdown right? How was this detected, valgrind?

Reply:
Yeah, only when shutting down ClientCallManager. I found it when writing the gRPC test in this PR: https://github.com/ray-project/ray/pull/17863. I tried to simulate client death in the test. but found the socket won't be closed when ClientCallManager and GrpcClient are deconstructed.  And finally found out that there is a memory leaking, the leaked call tags will hold the shared ptr of grpc channel, which cause the socket didn't close.

Reply:
Glad we are writing that test! :)